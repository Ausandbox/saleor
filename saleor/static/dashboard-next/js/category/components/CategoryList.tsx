import * as React from "react";
import Typography from "material-ui/Typography";
import Grid from "material-ui/Grid";
import Button from "material-ui/Button";
import { Link } from "react-router-dom";

import CategoryChildElement from "./CategoryChildElement";
import { categoryAddUrl, categoryShowUrl } from "../";
import { CategoryPropertiesQuery } from "../gql-types";
import i18n from "../../i18n";

interface CategoryListProps {
  categories?: CategoryPropertiesQuery["category"]["children"]["edges"];
}
export const CategoryList: React.StatelessComponent<CategoryListProps> = ({
  categories
}) => (
  <Grid container>
    {categories === undefined ? (
      <Grid item xs={12} sm={6} md={4} lg={3} xl={2}>
        <CategoryChildElement loading={true} label="" url="" />
      </Grid>
    ) : categories.length > 0 ? (
      categories.map(edge => (
        <Grid item xs={12} sm={6} md={4} lg={3} xl={2}>
          <CategoryChildElement
            url={categoryShowUrl(edge.node.id)}
            label={edge.node.name}
            key={edge.node.id}
          />
        </Grid>
      ))
    ) : (
      <Grid item xs={12}>
        <Typography variant="body2">{i18n.t("No categories found")}</Typography>
      </Grid>
    )}
  </Grid>
);

export default CategoryList;
