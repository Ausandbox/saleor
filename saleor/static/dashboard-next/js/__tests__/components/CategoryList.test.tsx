import * as React from "react";
import { MemoryRouter } from "react-router-dom";
import * as renderer from "react-test-renderer";

import CategoryList from "../../category/components/CategoryList";
import categoryListFixture from "./fixtures/categoryList";
import categoryFixture from "./fixtures/category";

describe("<CategoryList />", () => {
  it("renders while data is loading", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryList />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when data is fully loaded", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryList categories={categoryListFixture} />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
  it("renders when category list is empty", () => {
    const component = renderer.create(
      <MemoryRouter>
        <CategoryList categories={[]} />
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
