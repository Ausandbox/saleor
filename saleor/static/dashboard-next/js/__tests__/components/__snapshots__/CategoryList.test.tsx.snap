// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryList /> renders when category list is empty 1`] = `
<div
  className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-16-23"
>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38"
  >
    <aside
      className="MuiTypography-root-107 MuiTypography-body2-115"
    >
      No categories found
    </aside>
  </div>
</div>
`;

exports[`<CategoryList /> renders when data is fully loaded 1`] = `
<div
  className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-16-23"
>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38 MuiGrid-grid-sm-6-45 MuiGrid-grid-md-4-56 MuiGrid-grid-lg-3-68 MuiGrid-grid-xl-2-80"
  >
    <a
      className="MuiButtonBase-root-92 Component-button-91"
      href="/categories/Q2F0ZWdvcnk6MQ==/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiCardHeader-root-95"
      >
        <div
          className="MuiCardHeader-avatar-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-101"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </div>
        <div
          className="MuiCardHeader-content-98"
        >
          <span
            className="MuiTypography-root-107 MuiTypography-body2-115 MuiCardHeader-title-99"
          >
            Apparel
          </span>
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-132"
      />
    </a>
  </div>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38 MuiGrid-grid-sm-6-45 MuiGrid-grid-md-4-56 MuiGrid-grid-lg-3-68 MuiGrid-grid-xl-2-80"
  >
    <a
      className="MuiButtonBase-root-92 Component-button-91"
      href="/categories/Q2F0ZWdvcnk6Mg==/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiCardHeader-root-95"
      >
        <div
          className="MuiCardHeader-avatar-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-101"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </div>
        <div
          className="MuiCardHeader-content-98"
        >
          <span
            className="MuiTypography-root-107 MuiTypography-body2-115 MuiCardHeader-title-99"
          >
            Accessories
          </span>
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-132"
      />
    </a>
  </div>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38 MuiGrid-grid-sm-6-45 MuiGrid-grid-md-4-56 MuiGrid-grid-lg-3-68 MuiGrid-grid-xl-2-80"
  >
    <a
      className="MuiButtonBase-root-92 Component-button-91"
      href="/categories/Q2F0ZWdvcnk6Mw==/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiCardHeader-root-95"
      >
        <div
          className="MuiCardHeader-avatar-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-101"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </div>
        <div
          className="MuiCardHeader-content-98"
        >
          <span
            className="MuiTypography-root-107 MuiTypography-body2-115 MuiCardHeader-title-99"
          >
            Groceries
          </span>
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-132"
      />
    </a>
  </div>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38 MuiGrid-grid-sm-6-45 MuiGrid-grid-md-4-56 MuiGrid-grid-lg-3-68 MuiGrid-grid-xl-2-80"
  >
    <a
      className="MuiButtonBase-root-92 Component-button-91"
      href="/categories/Q2F0ZWdvcnk6NA==/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiCardHeader-root-95"
      >
        <div
          className="MuiCardHeader-avatar-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-101"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </div>
        <div
          className="MuiCardHeader-content-98"
        >
          <span
            className="MuiTypography-root-107 MuiTypography-body2-115 MuiCardHeader-title-99"
          >
            Books
          </span>
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-132"
      />
    </a>
  </div>
</div>
`;

exports[`<CategoryList /> renders while data is loading 1`] = `
<div
  className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-16-23"
>
  <div
    className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38 MuiGrid-grid-sm-6-45 MuiGrid-grid-md-4-56 MuiGrid-grid-lg-3-68 MuiGrid-grid-xl-2-80"
  >
    <a
      className="MuiButtonBase-root-92 Component-button-91"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="MuiCardHeader-root-95"
      >
        <div
          className="MuiCardHeader-avatar-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-101"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
            />
          </svg>
        </div>
        <div
          className="MuiCardHeader-content-98"
        >
          <span
            className="MuiTypography-root-107 MuiTypography-body2-115 MuiCardHeader-title-99"
          >
            <span
              className="Component-skeleton-131"
              style={
                Object {
                  "width": "80%",
                }
              }
            >
              â€Œ
            </span>
          </span>
        </div>
      </div>
      <span
        className="MuiTouchRipple-root-132"
      />
    </a>
  </div>
</div>
`;
